Vagrantfile

```diff
@@ -128,17 +128,35 @@ sudo iptables -t nat -A OUTPUT -d localhost -p tcp -m tcp --dport 53 -j REDIRECT
 
 echo "Pulling Docker images"
 
-find /vagrant/jobs -maxdepth 1 -type f -name '*.hcl' | xargs grep -E 'image\s*=\s*' | awk '{print $NF}' | sed -e 's/"//g' | while read j; do
+if [ -n "$DOCKERHUBID" ] && [ -n "$DOCKERHUBPASSWD" ]; then
+  echo "Login to Docker Hub as $DOCKERHUBID"
+  if ! echo "$DOCKERHUBPASSWD" | sudo docker login --username "$DOCKERHUBID" --password-stdin; then
+    echo 'Error login to Docker Hub'
+    exit 2
+  fi
+fi
+
+find /vagrant/jobs -maxdepth 1 -type f -name '*.hcl' | xargs grep -E 'image\s*=\s*' | awk '{print $NF}' | sed -e 's/"//g' -e 's/:demo//' | while read j; do
   echo "Pulling $j Docker image"
   if ! sudo docker pull $j >/dev/null; then
     echo "Exiting"
     exit 1
   fi
+  if ! echo "$j" | grep -q ':'; then
+    sudo docker tag "$j":latest "$j":demo
```

Comment:
What's that for ?

Reply:
Hello @cyriltovena

In the Nomad jobs files, you can decide to use a specific docker image version (like grafana 7.5.x) or the latest revision.

However as stated in [this PR comment](https://github.com/cyriltovena/observability-nomad/pull/6/commits/ef49947be74c10e34047ee1f68541e7a99eb852e) the behaviour is that for `latest` Nomad always do a force pull which will fail if you are hitting the rate limit.

The `trick/workaround` implemented here is to use your Docker Hub credentials to do the initial `docker pull` (from the Vagrantfile) and therefore pushing back the limits of rate limiting further down the path but as explained in the comment, the revision needs to be different from `latest`, therefore the usage of `demo`

As also explained in the commit comment, the other way to achieve this could be to prevent calling `docker logout` but in this case, the credentials will be left within the running image which is, I think, something which should be strongly avoided

HTH

Reply:
sounds good thank you !